ANMN/temperature_grid_product/anmn_temp_gridded_product.py

```diff
@@ -172,11 +172,39 @@ def _perdelta(start, end, delta):
 
 def create_time_1d(time_start, time_end, delta_in_minutes):
     """
-    create a 1D time array between start and end date and a data step of delta_in_minute
+    create a 1D time array between start and end date and a data step of delta_in_minutes
     """
     time_interp_array = []
-    for result in _perdelta(time_start + timedelta(minutes=delta_in_minutes)/2,
-                            time_end - timedelta(minutes=delta_in_minutes)/2,
+
+    # we modify the time_start and time_end values
+    if delta_in_minutes == 30:
+        if time_start.minute <= 15:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0)
+        elif time_start.minute > 15 and time_start.minute <= 45:
+            time_start = time_start.replace(minute=30, second=0, microsecond=0)
+        elif time_start.minute > 45:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+
+        if time_end.minute <= 15:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0)
+        elif time_end.minute > 15 and time_start.minute <= 45:
+            time_end = time_end.replace(minute=30, second=0, microsecond=0)
+        elif time_end.minute > 45:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+
+    elif delta_in_minutes == 60:
+        if time_start.minute > 30:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+        elif time_start.minte < 30:
+            time_start = time_start.replace(minute=0, second=0, microsecond=0)
+
+        if time_end.minute > 30:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0) + timedelta(hours=1)
+        elif time_end.minute < 30:
+            time_end = time_end.replace(minute=0, second=0, microsecond=0)
```

Comment:
There is a more generic way you can do this for any delta_in_minutes value. Your new time_start for example could be set by something like:
```Matlab
time_start_in_minutes = floor(time_start_in_minutes/delta_in_minutes)*delta_in_minutes
```